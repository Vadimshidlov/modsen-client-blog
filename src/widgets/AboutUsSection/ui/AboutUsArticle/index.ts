export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
