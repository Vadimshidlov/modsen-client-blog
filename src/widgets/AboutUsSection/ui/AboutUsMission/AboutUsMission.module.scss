.aboutUsMissionContainer {
  display: flex;
  column-gap: 32px;
  justify-content: space-between;
  padding: 128px 108px 64px;
  background: $lightPurple;
  margin-bottom: 128px;

  @media (max-width: $tablet) {
    margin-bottom: 64px;
    padding: 64px 54px 32px;
    flex-direction: column;
    row-gap: 20px;
  }

  @media (max-width: $mobile) {
    margin-bottom: 32px;
    padding: 32px 27px 16px;
  }

  .missionSubTitle {
    font-size: $fz16;
    letter-spacing: 3px;
    margin-bottom: 16px;
  }

  .missionTitle {
    margin-bottom: 16px;
    max-width: 481px;
    @include headingThree;

    @media (max-width: $tablet) {
      @include headingFive;
    }
  }

  .missionText {
    max-width: 516px;
    @include bodyOne;
  }
}
