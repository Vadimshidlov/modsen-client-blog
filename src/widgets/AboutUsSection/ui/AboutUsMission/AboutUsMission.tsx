import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission.module.scss';
import { Inter } from 'next/font/google';

const inter = Inter({
  subsets: ['latin'],
  weight: '600',
  variable: '--font-inter',
});

export function AboutUsMission() {
  return (
    <section className={styles.aboutUsMissionContainer}>
      <article>
        <h4 className={`${styles.missionSubTitle} ${inter.variable}`}>OUR MISSION</h4>
        <h3 className={styles.missionTitle}>
          Creating valuable content for creatives all around the world
        </h3>
        <span className={styles.missionText}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris
          in aliquam sem. At risus viverra adipiscing at in tellus.
        </span>
      </article>
      <article>
        <h4 className={`${styles.missionSubTitle} ${inter.variable}`}>OUR VISION</h4>
        <h3 className={styles.missionTitle}>A platform that empowers individuals to improve</h3>
        <span className={styles.missionText}>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris
          in aliquam sem. At risus viverra adipiscing at in tellus.
        </span>
      </article>
    </section>
  );
}
