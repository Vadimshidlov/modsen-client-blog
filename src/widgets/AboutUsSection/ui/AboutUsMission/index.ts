export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
