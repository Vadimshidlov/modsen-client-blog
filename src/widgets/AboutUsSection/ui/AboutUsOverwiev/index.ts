export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
