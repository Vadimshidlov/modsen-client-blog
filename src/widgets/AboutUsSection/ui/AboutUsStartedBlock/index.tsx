export { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock/AboutUsStartedBlock';
