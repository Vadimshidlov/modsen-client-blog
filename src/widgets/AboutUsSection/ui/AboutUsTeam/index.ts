export { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam/AboutUsTeam';
