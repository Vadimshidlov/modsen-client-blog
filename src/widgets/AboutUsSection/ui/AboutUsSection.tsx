import React from 'react';
import styles from '@/widgets/AboutUsSection/ui/AboutUsSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { AboutUsTitle } from '@/widgets/AboutUsSection/ui/AboutUsTitle';
import { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';
import { AboutUsTeam } from '@/widgets/AboutUsSection/ui/AboutUsTeam';
import { AboutUsStartedBlock } from '@/widgets/AboutUsSection/ui/AboutUsStartedBlock';

export function AboutUsSection() {
  return (
    <section className={`${styles.aboutUsSection} ${container.container}`}>
      <AboutUsTitle />
      <AboutUsOverwiev />
      <AboutUsMission />
      <AboutUsTeam />
      <AboutUsStartedBlock />
    </section>
  );
}
