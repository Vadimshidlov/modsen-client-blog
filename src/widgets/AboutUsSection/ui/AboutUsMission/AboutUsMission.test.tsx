import { render } from '@testing-library/react';
import { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission';

jest.mock('next/font/google', () => ({
  Inter: () => ({ variable: 'fake-variable' }),
}));

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('AboutUsMission component tests', () => {
  test('renders mission and vision sections with correct content', () => {
    const { getByText } = render(<AboutUsMission />);

    expect(getByText('Mocked missionTitle')).toBeInTheDocument();
    expect(getByText('Mocked visionTitle')).toBeInTheDocument();
  });

  test('should matches snapshot', () => {
    const { asFragment } = render(<AboutUsMission />);

    expect(asFragment()).toMatchSnapshot();
  });
});
