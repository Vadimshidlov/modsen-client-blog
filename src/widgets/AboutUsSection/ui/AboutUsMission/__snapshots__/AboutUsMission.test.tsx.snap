// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUsMission component tests should matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="aboutUsMissionContainer"
  >
    <article>
      <h4
        class="missionSubTitle fake-variable"
      >
        Mocked missionSubTitle
      </h4>
      <h3
        class="missionTitle"
      >
        Mocked missionTitle
      </h3>
      <span
        class="missionText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.
      </span>
    </article>
    <article>
      <h4
        class="missionSubTitle fake-variable"
      >
        Mocked visionSubTitle
      </h4>
      <h3
        class="missionTitle"
      >
        Mocked visionTitle
      </h3>
      <span
        class="missionText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.
      </span>
    </article>
  </section>
</DocumentFragment>
`;
